tests/testthat/test-new-blog-post.R

```diff
@@ -0,0 +1,68 @@
+test_that("Create a blog post", {
+  skip_if_no_quarto("1.4")
+
+  current_dir <- getwd()
+
+  temp_dir <- withr::local_tempdir()
+  dir_path <- fs::path(temp_dir, "test-blog-project")
+
+  withr::defer(fs::dir_delete(dir_path), envir = rlang::current_env())
+
+  quarto_create_project(name = "test-blog-project", type = "blog",
+                        dir = temp_dir, quiet = TRUE)
+
+  setwd(dir_path)
+  withr::defer(setwd(current_dir), envir = rlang::current_env())
```

Comment:
Usually I use `withr::local_dir()` to change working directory for the rest of the test

This should be enough 
````r
withr::local_dir(fs::path(temp_dir, "test-blog-project"))
````